addons/binding/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/handler/KeContactP20Handler.java

```diff
@@ -0,0 +1,967 @@
+				case "Sec": {
+					long state = entry.getValue().getAsLong();
+
+					Calendar uptime = Calendar.getInstance();
+					uptime.setTimeZone(TimeZone.getTimeZone("GMT"));
+					uptime.setTimeInMillis(state * 1000);
+					SimpleDateFormat pFormatter = new SimpleDateFormat(
+							"yyyy-MM-dd'T'HH:mm:ss");
+					pFormatter.setTimeZone(TimeZone.getTimeZone("GMT"));
+
+					updateState(
+							new ChannelUID(getThing().getUID(), CHANNEL_UPTIME),
+							new DateTimeType(
+									pFormatter.format(uptime.getTime())));
+					break;
+				}
+				case "U1": {
+					int state = entry.getValue().getAsInt();
+					State newState = new DecimalType(state);
+					updateState(
+							new ChannelUID(getThing().getUID(), CHANNEL_U1),
+							newState);
+					break;
+				}
+				case "U2": {
+					int state = entry.getValue().getAsInt();
+					State newState = new DecimalType(state);
+					updateState(
+							new ChannelUID(getThing().getUID(), CHANNEL_U2),
+							newState);
+					break;
+				}
+				case "U3": {
+					int state = entry.getValue().getAsInt();
+					State newState = new DecimalType(state);
+					updateState(
+							new ChannelUID(getThing().getUID(), CHANNEL_U3),
+							newState);
+					break;
+				}
+				case "I1": {
+					int state = entry.getValue().getAsInt();
+					State newState = new DecimalType(state / 1000);
+					updateState(
+							new ChannelUID(getThing().getUID(), CHANNEL_I1),
+							newState);
+					break;
+				}
+				case "I2": {
+					int state = entry.getValue().getAsInt();
+					State newState = new DecimalType(state / 1000);
+					updateState(
+							new ChannelUID(getThing().getUID(), CHANNEL_I2),
+							newState);
+					break;
+				}
+				case "I3": {
+					int state = entry.getValue().getAsInt();
+					State newState = new DecimalType(state / 1000);
+					updateState(
+							new ChannelUID(getThing().getUID(), CHANNEL_I3),
+							newState);
+					break;
+				}
+				case "P": {
+					long state = entry.getValue().getAsLong();
+					State newState = new DecimalType(state / 1000);
+					updateState(new ChannelUID(getThing().getUID(),
+							CHANNEL_POWER), newState);
+					break;
+				}
+				case "PF": {
+					int state = entry.getValue().getAsInt();
+					State newState = new PercentType(state / 10);
+					updateState(new ChannelUID(getThing().getUID(),
+							CHANNEL_POWER_FACTOR), newState);
+					break;
+				}
+				case "E pres": {
+					long state = entry.getValue().getAsLong();
+					State newState = new DecimalType(state / 10);
+					updateState(new ChannelUID(getThing().getUID(),
+							CHANNEL_SESSION_CONSUMPTION), newState);
+					break;
+				}
+				case "E total": {
+					long state = entry.getValue().getAsLong();
+					State newState = new DecimalType(state / 10);
+					updateState(new ChannelUID(getThing().getUID(),
+							CHANNEL_TOTAL_CONSUMPTION), newState);
+					break;
+				}
+				}
+			}
+
+		} catch (JsonParseException e) {
+			logger.warn("Some invalid JSON data will be ignored: '{}'",
```

Comment:
Some or all? If it is "some" and this is ok to ignore it, reduce the logging level to debug. If this means a serious problem, point to it with the logging message.